client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/netty/ChannelFactory.java

```diff
@@ -110,23 +112,30 @@ public class ChannelFactory {
    private CompletableFuture<Void> clusterSwitchStage;
    // Servers for which the last connection attempt failed and which have no established connections
    @GuardedBy("lock")
-   private final Set<SocketAddress> failedServers = new HashSet<>();
+   private final Set<SocketAddress> failedServers;
+   private final Configuration configuration;
    private final CodecHolder codecHolder;
    private AddressResolverGroup<?> dnsResolver;
    private SslContext sslContext;
    private FileWatcher watcher;
 
-   public ChannelFactory(CodecHolder codecHolder) {
+   public ChannelFactory(Configuration configuration, CodecHolder codecHolder) {
+      this.configuration = configuration;
       this.codecHolder = codecHolder;
+
+      this.failedServers = configuration.clientIntelligence() == ClientIntelligence.BASIC ?
+            Collections.newSetFromMap(Caffeine.newBuilder()
```

Comment:
`if (configuration.basicFailedServerTimeout() > 0)`  Allows the user to turn it off

Reply:
Sure, sounds fine good to me.